Automattic/woocommerce-payments
# Add ToS modal

Fixes #897 

This PR is only one of the ToS acceptance handling pieces. With current implementation it will never be loaded to the client or displayed to the user. Logic triggering and enqueuing the script need to be added to make it work for the users.

#### Changes proposed in this Pull Request

* Add new js entry point to load ToS scripts independently.
* Add modal UI to capture ToS.
* Add controller to handle ToS acceptance requests without real ToS acceptance tracking.
* Disable plugin when ToS are declined.


#### Testing instructions

* Go to `includes/admin/class-wc-payments-admin.php` and remove `false &&` condition on line 280. It prevents script from loading to the client.
* Go to WCPay seettings page. The ToS modal should appear. 
* The modal should behave as described in paJDYF-R2-p2.
* When clicking on `disable` button the modal should close, user have to be redirected to the payment gateway page and the gateway should be disabled.


**Note on changelog entry**: Not adding at the moment as feature is not ready for general usage.

-------------------
